erpnext/stock/report/stock_balance/stock_balance.py

```diff
@@ -257,6 +257,9 @@ def get_item_reorder_details(items):
 	return dict((d.parent + d.warehouse, d) for d in item_reorder_details)
 
 def validate_filters(filters):
+	if filters.warehouse_type and not filters.warehouse:
+		frappe.throw(_("Please set Warehouse"))
```

Comment:
Don't make warehouse mandatory. Instead, consider all warehouses with the selected warehouse type.

Reply:
done